File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/adapter/HeaderAndFooterAdapter.java
package com.chad.baserecyclerviewadapterhelper.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.chad.baserecyclerviewadapterhelper.R;
import com.chad.baserecyclerviewadapterhelper.databinding.ItemHeaderAndFooterBinding;
import com.chad.baserecyclerviewadapterhelper.entity.Status;
import com.chad.library.adapter.base.BaseQuickAdapter;

import java.util.List;

/**
 * https://github.com/CymChad/BaseRecyclerViewAdapterHelper
 */
public class HeaderAndFooterAdapter extends BaseQuickAdapter<Status, HeaderAndFooterAdapter.VH> {

    public HeaderAndFooterAdapter(List<Status> list) {
        super(list);
    }

    static class VH extends RecyclerView.ViewHolder {

        ItemHeaderAndFooterBinding binding;

        public VH(@NonNull ItemHeaderAndFooterBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    @NonNull
    @Override
    protected HeaderAndFooterAdapter.VH onCreateViewHolder(@NonNull Context context, @NonNull ViewGroup parent, int viewType) {
        ItemHeaderAndFooterBinding binding = ItemHeaderAndFooterBinding.inflate(LayoutInflater.from(context), parent, false);
        return new VH(binding);
    }

    @Override
    protected void onBindViewHolder(@NonNull HeaderAndFooterAdapter.VH holder, int position, Status item) {
        switch (holder.getLayoutPosition() % 3) {
            case 0:
                holder.binding.iv.setImageResource(R.mipmap.animation_img1);
                break;
            case 1:
                holder.binding.iv.setImageResource(R.mipmap.animation_img2);
                break;
            case 2:
                holder.binding.iv.setImageResource(R.mipmap.animation_img3);
                break;
            default:
                break;
        }
    }

}

File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/activity/headerfooter/adapter/HeaderAndFooterAdapter.kt
package com.chad.baserecyclerviewadapterhelper.activity.headerfooter.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.chad.baserecyclerviewadapterhelper.R
import com.chad.baserecyclerviewadapterhelper.databinding.ItemHeaderAndFooterBinding
import com.chad.baserecyclerviewadapterhelper.entity.Status
import com.chad.library.adapter.base.BaseQuickAdapter

/**
 * https://github.com/CymChad/BaseRecyclerViewAdapterHelper
 */
class HeaderAndFooterAdapter(list: List<Status>) :
    BaseQuickAdapter<Status?, HeaderAndFooterAdapter.VH>(list) {

    class VH(var binding: ItemHeaderAndFooterBinding) : RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(context: Context, parent: ViewGroup, viewType: Int): VH {
        val binding =
            ItemHeaderAndFooterBinding.inflate(LayoutInflater.from(context), parent, false)
        return VH(binding)
    }

    override fun onBindViewHolder(holder: VH, position: Int, item: Status?) {
        when (holder.layoutPosition % 3) {
            0 -> holder.binding.iv.setImageResource(R.mipmap.animation_img1)
            1 -> holder.binding.iv.setImageResource(R.mipmap.animation_img2)
            2 -> holder.binding.iv.setImageResource(R.mipmap.animation_img3)
            else -> {}
        }
    }
}